app/controllers/agents_controller.rb

```diff
@@ -50,7 +50,12 @@ def dry_run
     agent.name ||= '(Untitled)'
 
     if agent.valid?
-      results = agent.dry_run!
+      if event_payload = params[:event]
+        dummy_agent = Agents::ManualEventAgent.new(name: 'Dry-Runner')
+        event = dummy_agent.events.build(user: current_user, payload: event_payload)
```

Comment:
Good point!  I know `event` will never be saved before `dummy_agent` can be saved, which cannot when it is missing `user`, but I'll make that clear.